File: aroundHubSpringBoot/src/main/java/com/example/aroungHubJavaSpringBoot/aroundHubSpringBoot/data/handler/impl/ProductDataHandlerImpl.java
package com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.handler.impl;

import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.dao.ProductDAO;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.entity.ProductEntity;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.handler.ProductDataHandler;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class ProductDataHandlerImpl implements ProductDataHandler {

    ProductDAO productDAO;

    @Autowired
    public ProductDataHandlerImpl(ProductDAO productDAO){
        this.productDAO = productDAO;
    }

    @Override
    public ProductEntity saveProductEntity(String productId, String productName, int productPrice, int productStock) {
        ProductEntity productEntity = new ProductEntity(productId, productName, productPrice, productStock);

        return productDAO.saveProduct(productEntity);
    }

    @Override
    public ProductEntity getProductEntity(String productId) {
        return productDAO.getProduct(productId);
    }
}

File: aroundHubKotlinSpringBoot/src/main/kotlin/com/example/aroungHubKotlinSpringBoot/aroundHubKotlinSpringBoot/data/handler/impl/ProductDataHandlerImpl.kt
package com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.handler.impl

import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.dao.ProductDAO
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.entity.ProductEntity
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.handler.ProductDataHandler
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class ProductDataHandlerImpl(
    var productDAO : ProductDAO
) : ProductDataHandler{
    override fun saveProductEntity(
        productId: String,
        productName: String,
        productPrice: Int,
        productStock: Int
    ): ProductEntity {
        val productEntity = ProductEntity(
            productId = productId,
            productName = productName,
            productPrice = productPrice,
            productStock = productStock
        )
        return productDAO.saveProduct(productEntity)
    }

    override fun getProductEntity(productId: String): ProductEntity {
        return productDAO.getProduct(productId)
    }
}